Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -3026,36 +3027,59 @@ namespace OpenInfraPlatform
 				{
 
 					std::vector<typename IfcEntityTypesT::IfcReal> coefficientX, coefficientY, coefficientZ;
-					typename IfcEntityTypesT::IfcReal x, y, z;
+					//typename IfcEntityTypesT::IfcReal x, y, z;
+					double x, y, z;
 					int polynomialConstantCntX, polynomialConstantCntY, polynomialConstantCntZ;
+					std::vector<double> polynomialConstantX, polynomialConstantY, polynomialConstantZ;
 
 					if (polynomialCurve->CoefficientsX)
 					{
 						// Interpret coefficients for X
 						coefficientX = polynomialCurve->CoefficientsX;
 						polynomialConstantCntX = std::size(coefficientX);
-						// Implement polynomial term for x coordinate
-						x = calculatePolynomialCurve(coefficientX, parameter);
+						//std::vector<double> polynomialConstantX;
+						//convert to double
+						for (int i = 0; i < std::size(coefficientX); i++)
+						{
+							polynomialConstantX.push_back(double(coefficientX[i]));
+						}

```

Comment:
Learning possibility: look up std::transform function, which could help you achieve the same with less code. (there are some usages in OIP code already)

Reply:
Resolved: [98adeb1](https://github.com/tumcms/Open-Infra-Platform/pull/539/commits/98adeb1aa6baac4fd3b81995623cb7bff780ddb0)